lib/linguist/strategy/modeline.rb

```diff
@@ -49,20 +49,20 @@ class Modeline
         # serves as a terminator for an option sequence, delimited by whitespace.
         (?=
           # So we have to ensure the modeline ends with a colon
-          : (?=\s* set? \s [^\n:]+ :) |
+          : (?=[[:blank:]]* set? [[:blank:]] [^\n:]+ :) |
```

Comment:
I'd recommend the use of `[ \t]` instead, as it's arguably clearer than a POSIX character class.